### PR TITLE
fix(triton_attn): support decoupled scale caches to preserve 128B memory alignment

### LINKED ISSUE #56081: Per-token-head inline KV scales break 128B alignment — 6.3× slowdown in gather-based attention kernels
### Summary

`TritonAttentionBackend` packs the per-`(token, head)` fp32 scale **inline, immediately
after each head's quantized data**:

```
[ K data (hs) | K scale (pad) | V data (hs) | V scale (pad) ]
```

with `pad = sizeof(float32) / sizeof(cache_dtype)`
(`TritonAttentionBackend.get_kv_cache_shape`, `customize_spec`, `_ensure_scale_caches`).

For `head_size=256` with an int8 cache this makes the per-head slot **260 bytes** and the
K/V pair **520 bytes**. Neither is a multiple of 128.

For kernels that walk the KV cache **contiguously** this is fine. For kernels that
**gather scattered tokens** — sparse / selective attention, where each loaded token comes
from an arbitrary `(block, offset)` — every token load starts at a non-128B-aligned
address. On GB10 (sm_121) this costs **6.3×** in the attention kernel.

### Measurement

Qwen3.8-Flash-Next QSA sparse paged attention (`qsa_sparse_paged_attention`), GB10 sm_121,
2048 query rows, topk 1024, `head_size=256`, 2 KV heads, page size 64. Identical kernel,
identical data, only the cache **stride** differs:

| Layout | Row stride | Kernel time | vs contiguous |
|---|---|---|---|
| contiguous int8 | 256 B | **9.6 ms** | — |
| inline scale (as allocated today) | 260 B | **60.2 ms** | **+526 %** |
| padded to 128B alignment (slot 640 B) | 320 B | 12.7 ms | +32 % |
| fp8 per-tensor (no inline scale) | 256 B | 15.9 ms | +65 % |

The only difference between rows 1 and 2 is where the scales live. The values, the
quantization, and the kernel are the same.

Padding the slot up to a 128B multiple recovers most of the loss but costs 20 % of KV
capacity (640 B vs 512 B per slot), which defeats the point of quantizing.

### Workaround

Allocate the scales in a **separate tensor** instead of inline:

```python
self._k_scale_cache = torch.ones(
    num_blocks, block_size, num_kv_heads, dtype=torch.float32, device=...
)
```

The KV data then keeps its natural 512 B stride and stays aligned. Memory overhead is
unchanged — 2 × 4 B per (token, head) either way, ~1.6 % of the 1024 B/token data for this
shape. With this change the same kernel runs at 9.6 ms.

It also removes the need to override `customize_spec` / `get_kv_cache_shape`: the cache
shape becomes identical to the non-per-token-head case, and the resulting KV pool size
matches fp8 exactly (857,210 tokens in our configuration).

### Suggestion

Two options, either would help:

1. **Document the constraint** near `customize_spec` — that inline scales assume
   contiguous KV access and are not suitable for gather-based backends.
2. **Make scale placement selectable per backend**, e.g. a class attribute on
   `AttentionBackend` that opts out of inline scales, so gather-based backends can use a
   side tensor without reimplementing the allocation path.

I would be glad to send a PR for either if there is a preference on the shape of the fix.

### Context

Found while adding `int8_per_token_head` support to the QSA sparse-attention path of
Qwen3.8-Flash-Next (related: #54426, #55557). The initial end-to-end result looked like a
40 % prefill regression and I nearly abandoned int8 over it; isolating the kernel showed
int8 was in fact **faster** than both fp8 and bf16, and the entire regression came from
this layout. After switching to a side scale tensor, int8 beats fp8 on every axis at an
identical KV pool size (TTFT −7.9 %, decode +5.2 %, MTP acceptance +1.9 pts over a 9-cell
grid with matched background load).

The full patch set, the probe that produced the table above, and the raw benchmark grids
are in [rmagur1203/vllm-qsa-int8#1](https://github.com/rmagur1203/vllm-qsa-int8/pull/1):
`qsa-kv-int8/tools/qsa_align_probe.py` reproduces the four rows with nothing but the
kernel and a stride change, and `qsa-kv-int8/patches/wire_qsa_int8_kv.py` is the
side-tensor workaround as deployed.

### Environment

- GB10 / DGX Spark, sm_121, driver 590.48.01, aarch64
- vLLM `0.1.dev20073+g8e685d198` (pinned Qwen3.8-Flash-Next preview image)
- The affected code (`triton_attn.py` per-token-head layout) is unchanged on `main`.

⚠️ I could not reproduce on `main`: PLE CPU offload is not upstream, so this 125B
checkpoint does not load on a single 121 GiB GB10 without the preview image. The layout
code itself is identical on `main`, so the reasoning should carry, but the numbers above
are from the preview build.

### PR DESCRIPTION
**Pull request by debanshd**
Fixes #56081.

### Description

In `TritonAttentionBackend`, per-token-head quantized KV scales are packed inline by default, appending an fp32 scale after each head's quantized data. For `head_size=256` int8 caches, each head slot becomes 260 bytes (520 bytes per K/V pair). This breaks 128-byte memory coalescing, leading to a 6.3x slowdown in gather-based and sparse attention kernels (such as QSA sparse paged attention).

This PR makes scale cache placement configurable per backend and layer:
1. Adds `supports_inline_scales: ClassVar[bool] = True` on `AttentionBackend` and `TritonAttentionBackend`.
2. Documents the alignment constraint in `customize_spec` and skips inline scale padding when `supports_inline_scales` is False, retaining natural unpadded 512-byte slot layout.
3. In `TritonAttentionImpl._ensure_scale_caches`, allocates decoupled side-tensor scale caches (`_k_scale_cache` and `_v_scale_cache`) with shape `(num_blocks, block_size, num_kv_heads)` in float32 when `supports_inline_scales` is False.
4. In `TritonAttentionImpl._pth_key_value_caches`, splits unpadded caches cleanly into 128-byte aligned key and value slices.
5. In `Attention` layer, forwards `supports_inline_scales` from backend to implementation.
6. Adds comprehensive regression unit tests in `tests/v1/attention/test_triton_kv_scale_alignment.py`.

### CODE REVIEW

**Comment by debanshd** [comment]
Hi @tdoublep @LucasWilkinson @MatthewBonanni, could you please take a look at this fix for 128B memory alignment when using FP8 KV cache scales in `triton_attn.py` (#56081) when you have a moment? All unit tests and DCO checks are passing.
